src/components/PeoplePage.tsx

```diff
@@ -0,0 +1,123 @@
+import { useEffect, useState } from "react";
+import { Person } from "../types";
+import { getPeople } from "../api";
+import { Loader } from "./Loader";
+import { Link, useParams } from "react-router-dom";
+import classNames from "classnames";
+
+
+export const PeoplePage = () => {
+
+  const [people, setPeople] = useState<Person[]>([]);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [error, setError] = useState<string | null>(null);
+
+  const { personSlug } = useParams();
+
+  const personInfo = ['Name', 'Sex', 'Born', 'Died', 'Mother', 'Father'];
+
+  const personName = (name: string | null) => {
+    return people.find(person => person.name === name);
+  }
+
+
+  useEffect(() => {
+   setLoading(true)
+   getPeople()
+   .then(data => {
+    setPeople(data)
+    setLoading(false)
+   })
+   .catch(err => {
+     setError('Failed to load people data')
+     setLoading(false)
+   })
+  }, [])
+
+
+  if (loading) {
+    return <Loader />;
+  }
+
+  if (error) {
+    return (
+      <p data-cy="peopleLoadingError" className="has-text-danger">
+        Something went wrong
+      </p>
+    );
+  }
+
+  if (!people.length) {
+    return <p data-cy="noPeopleMessage">There are no people on the server</p>;
+  }
```

Comment:
This check for 'people.length' is redundant because if the data fetching fails or if there are no people on the server, you are already returning from the component before this check. So, you can safely remove this check.